### PR TITLE
chore(documentation-site): NO-JIRA enhance sidebar

### PR DESCRIPTION
**Pull request by juliodialpad**
# Enhance sidebar

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will not increment the version number, but will still deploy to documentation site on release:

- [ ] Documentation
- [ ] Build system
- [ ] CI
- [ ] Style (code style changes, not css changes)
- [ ] Test
- [x] Other (chore)

## :book: Description

- Enhanced the sidebar with recursive capacities to be able to nest children within children.
- Moved /content to /guides/content.
- Added the ability to expand/collapse categories.

## :bulb: Context

Nested categories requested by Francis.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have considered the performance impact of my change.

### CODE REVIEW

**Comment by francisrupert** [comment]
Awesome! Good use of Collapsible.

I'll list some comments below as individual comments rather than one long one:

**Comment by francisrupert** [comment]
The [Content page](https://dialtone.dialpad.com/deploy-previews/pr-406/guides/content/) should have its own grid of child items listed too, like the attached faked screenshot below.

Those same items should not appear on the [Guides landing page](https://dialtone.dialpad.com/deploy-previews/pr-406/guides/).

image

**Comment by francisrupert** [comment]
When arriving on the [Guides page](https://dialtone.dialpad.com/deploy-previews/pr-406/guides/), the **Content** item should remain collapsed.

**Comment by francisrupert** [comment]
The **Content** left-nav item should be a single target. Clicking anywhere on the row expands and navigates to that page, it does not collapse.

**Comment by francisrupert** [comment]
The "**Content**" item should be same text style as its peers. 

image

**Comment by francisrupert** [comment]
At the moment, we only want items that have children to be a Collapsible. That is, only "Content" on [Guides](https://dialtone.dialpad.com/deploy-previews/pr-406/guides/) should be collapsible.

The following left-nav titles should **_not_** be collapsible:

* "Guides" on [Guides](https://dialtone.dialpad.com/deploy-previews/pr-406/guides/).
* "Components" on [Components](https://dialtone.dialpad.com/deploy-previews/pr-406/components/).
* "Design language" on [Design language](https://dialtone.dialpad.com/deploy-previews/pr-406/design/).
* Each left-nav section on [Utilities](https://dialtone.dialpad.com/deploy-previews/pr-406/utilities/).
* "Design Tokens" on [Tokens](https://dialtone.dialpad.com/deploy-previews/pr-406/tokens/).
* "About" on [About](https://dialtone.dialpad.com/deploy-previews/pr-406/about/dialtone.html).

As an example, arriving on [Guides](https://dialtone.dialpad.com/deploy-previews/pr-406/guides/) should appears like so...

image

**Comment by hynes-dialpad** [comment]
I agree with @francisrupert. Other than that, this LGTM!

**Review by francisrupert — DISMISSED** [review]
Officially marking as "request changes", refer to standalone comments above.

**Comment by francisrupert** [comment]
Remove the `d-pl8` from the first level. It should only be there for the nested one. 

| 👎 | 👍 |
|--------|--------|
| image | image |

**Comment by francisrupert** [comment]
On Guides' landing page, only the items seen in Guides' leftnav should appear as a card.

image

**Comment by francisrupert** [comment]
The Content section should be collapsed if the current page is not within Content section.

https://github.com/user-attachments/assets/c9e5cc06-450c-45db-8fc6-9ac350e5be1c

**Comment by francisrupert** [comment]
The leftnav should not expand when arriving at a page. Perhaps this is a limitation of DtCollapsible, but it appears to be collapsed onload and then expand onload.

https://github.com/user-attachments/assets/f70e65e2-f4d0-41ff-a15b-340705d26c6d

**Comment by juliodialpad** [comment]
> The leftnav should not expand when arriving at a page. Perhaps this is a limitation of DtCollapsible, but it appears to be collapsed onload and then expand onload.
> 
>  Screen.Recording.2024-07-12.at.3.48.47.PM.mov

Thought it'd not collapse once opened by this comment

> The **Content** left-nav item should be a single target. Clicking anywhere on the row expands and navigates to that page, it does not collapse.

Misunderstood the behavior, fixing it 😅

File: apps/dialtone-documentation/docs/.vuepress/theme/components/Sidebar.vue [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <dt-stack
     v-if="sidebarItems[0]?.children.length"
     as="ul"
-    gap="500"
```
Comment (hynes-dialpad) [inline apps/dialtone-documentation/docs/.vuepress/theme/components/Sidebar.vue]: On the components page, there should be a gap of 500 between the sections.

**Current**
image

**Proposed Update**
image

**Review by hynes-dialpad — CHANGES_REQUESTED** [review]
One small change here to increase the space between sections between sections on the Utilities page.

**Comment by francisrupert** [comment]
The "Previous|Next" pagination doesn't exist. Unless it's a super quick fix, can probably just log a task to fix later.

image

**Comment by francisrupert** [comment]
I think there's a backlog item for better RWD view of all navigation, but this ain't pretty. If you can't find a backlog item, create it. 

image

**Comment by fede-dp** [comment]
This looks 100% better, great job!
one small comment here:
 this one is bold
64f907820bb7a964e2f3429d4e57720e
and this one isn't
7d0ab10cc86f7288650694d9e91ef6e2
is this on purpose?
because if we open the main 'content' tab, that one should be bold like every other tab, right? or we keep it that way for something in particular? in their child options works fine but not in the main category.

**Comment by juliodialpad** [comment]
> The "Previous|Next" pagination doesn't exist. Unless it's a super quick fix, can probably just log a task to fix later.
> 
> image

Fixed

File: apps/dialtone-documentation/docs/.vuepress/theme/components/Sidebar.vue (hunk 1 above)
Comment (juliodialpad) [reply apps/dialtone-documentation/docs/.vuepress/theme/components/Sidebar.vue]: Fixed

**Comment by juliodialpad** [comment]
> This looks 100% better, great job! one small comment here: this one is bold 64f907820bb7a964e2f3429d4e57720e and this one isn't 7d0ab10cc86f7288650694d9e91ef6e2 is this on purpose? because if we open the main 'content' tab, that one should be bold like every other tab, right? or we keep it that way for something in particular? in their child options works fine but not in the main category.

Thanks for catching that! Just fixed it.

**Comment by hynes-dialpad** [comment]
@juliodialpad Tertiary menu items are using `d-fw-semi-bold`: 
image

Let's remove that utility class. The indentation is sufficient IMO
image

**Comment by juliodialpad** [comment]
> I think there's a backlog item for better RWD view of all navigation, but this ain't pretty. If you can't find a backlog item, create it.
> 
> image

Found a ticket but not quite sure it's exactly what you're mentioning: https://dialpad.atlassian.net/browse/DLT-895

Screenshot 2024-07-15 at 8 12 00 p m

I did this fix in the meantime.

**Comment by juliodialpad** [comment]
> @juliodialpad Tertiary menu items are using `d-fw-semi-bold`: image
> 
> Let's remove that utility class. The indentation is sufficient IMO image

Just pushed a fix that might address that.

**Comment by hynes-dialpad** [comment]
@juliodialpad I still was seeing the font weight so I went ahead and make the corrections myself. Please review and let me know.

**Review by hynes-dialpad — APPROVED** [review]

**Pull request merged**